### PR TITLE
[Fix] Error handling send email notification #179

### LINKED ISSUE #179: [bug] Add error handling in send_email_notification handler
If a notification setting does not exist in the database, the following code will raise `DoesNotExist` exception. 
https://github.com/openwisp/openwisp-notifications/blob/a20957b086fa10ab49a4488a231c3e1c926b617a/openwisp_notifications/handlers.py#L173-L177

**Possible solution**

If such an exception is raised, return from the function i.e. don't send any email

**Notes for writing failing tests**

1. Ensure the recipient of the notification does not have any NotificationSetting
2. Instead of using `self._create_notification` helper of the `TestNotifications` class create a notification using `Notification.objects.create()`. The former will not create a notification if NotificationSetting does not exist in the database

### PR DESCRIPTION
**Pull request by aagman945**
Added exception handling for the issue #179 and test for the same . So ,If throws exception the function will be returned empty and mail will not be sent.

### CODE REVIEW

File: openwisp_notifications/tests/test_notifications.py [hunk 1]
```diff
@@ -851,3 +852,28 @@ def test_notification_cache_update(self):
...
         self.assertEqual(operator_cache.username, 'new operator name')
+
+class test_email_notification(TestOrganizationMixin, TransactionTestCase):
+    def setUp(self):
+        self.user=self._create_user()
+        self.admin = self._create_admin()
+        self.target_obj = self._get_org_user()
+        self.data = dict(
+            email_subject='Test Email subject', url='https://localhost:8000/admin'
+        )
```
Comment (pandafy) [inline openwisp_notifications/tests/test_notifications.py:863]: We don't need to add a dedicated test class here. You can add this test to `TestTransactionNotifications` class. 

P.S.: The python convention is to use PascalCase for naming classes not snake_case. Refer https://namingconvention.org/python/

**Review by pandafy — CHANGES_REQUESTED** [review]
@aagman945 kindly **read the contributing guidelines** and update your editor settings as mentioned here https://openwisp.io/docs/developer/contributing.html#avoiding-unnecessary-changes. 

Doing so would have reduced requested changes in this PR. The commit message needs to be updated as well. 

Also, always run `./run-qa-check` before pushing changes to GitHub.

**Comment by aagman945** [comment]
@pandafy ./run-qa-checks throws error 
TypeError: [files].flat is not a function
at standalone (/usr/local/lib/node_modules/stylelint/lib/standalone.js:158:25)
at module.exports (/usr/local/lib/node_modules/stylelint/lib/cli.js:484:9)
at Object.<anonymous> (/usr/local/lib/node_modules/stylelint/bin/stylelint.js:8:22)
at Module._compile (internal/modules/cjs/loader.js:778:30)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
at Module.load (internal/modules/cjs/loader.js:653:32)
at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
at Function.Module._load (internal/modules/cjs/loader.js:585:3)
at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
at startup (internal/bootstrap/node.js:283:19)

File: openwisp_notifications/tests/test_notifications.py [hunk 2]
```diff
@@ -813,6 +813,23 @@ def test_delete_obsolete_tasks(self, mocked_task, *args):
...
+            recipient=self.admin,
+            description='Test Notification Description',
+            verb='Test Notification',
+            action_object=target_obj,
+            target=target_obj,
+            data=data,
+            type="default",
+        )
+        self.assertEqual(len(mail.outbox), 1)
+
```
Comment (pandafy) [inline openwisp_notifications/tests/test_notifications.py:832]: @aagman945 this test does not trigger the bug. The bug only appears if notification setting is not present for the  user. 

In the end of the test, you are assert for presence of one email. But we don't want an email to be sent in this scenario. This depicts that the test does not tests the exception handling that  you just added. 

I would change this test as following:

```suggestion
def test_email_notification_without_notification_setting(self):
        target_obj = self._get_org_user()
        data = dict(
            email_subject='Test Email subject', url='https://localhost:8000/admin'
        )
        # Delete notification settings of the recipient
        self.admin.notificationsetting_set.all().delete()
        Notification.objects.create(
            actor=self.admin,
            recipient=self.admin,
            description='Test Notification Description',
            verb='Test Notification',
            action_object=target_obj,
            target=target_obj,
            data=data,
            type="default",
        )
        self.assertEqual(len(mail.outbox), 0)
```

**Review by pandafy — CHANGES_REQUESTED** [review]

**Review by pandafy — APPROVED** [review]

**Pull request merged**